### ISSUE TITLE
Reduce the ceremony of using worker API 

### ISSUE THREAD

**Issue by pioterj** [gradle/gradle#7032]
Currently, [using worker API](https://guides.gradle.org/using-the-worker-api/) requires some ceremony:
- Implement `Runnable`
- Inject `WorkerExecutor` to a task
- Submit `Runnable` implementation to the executor and configure the worker

It would be nice to make creating a task to be run in parallel with less code.

For example, we could provide an ability to annotate the `TaskAction` with some additional annotation, like `@ParallelEnabled`. Gradle could then wrap the call to the task action in a `Runnable`. Parameters on the annotation could be used for setting the isolation levels. While it may not be a feasible approach for breaking up work in a single task, it would make it easier to achieve intra-project parallelism with monolith custom tasks.

**Comment by melix** [comment]
We shouldn't assume any particular implementation. There are lots of cases which are far from trivial: automatic isolation of parameters, usage of services, static fields, task classpath inference, not leaking runtime to the worker, ...

**Comment by pioterj** [comment]
Absolutely. The last paragraph is just an example of how the ceremony could be potentially addressed.

**Comment by lptr** [comment]
This is what the isolated tasks proposal is supposed to address.

**Comment by dcabasson** [comment]
@melix I guess what I would be interested in is a simple implementation for trivial cases. I agree that there are more complex cases, and those can leverage the current approach.
But for trivial cases such as the one in the example guide, it would be nice to have a simple configuration to enable parallel execution of tasks in a project.

**Comment by melix** [comment]
No disagreement on my side. Just saying that "simple" is not that simple, because for the "simple" cases we already have to validate some assumptions, like the property types, so that we can isolate the inputs. Otherwise it's just going to fail badly, which is not very nice UX :)

**Comment by dcabasson** [comment]
True - I guess you can put a big disclaimer of "Use at your own risks". But know that there are users (such as me) who would be ready to use it even if it requires to thoroughly test the changes in our plugins.

**Comment by mathjeff** [comment]
As a user, I think some easier way to enable the parallel execution of multiple tasks in one project as requested in this issue would still be really nice (dear staleness bot, this seems non-stale to me even if there isn't bandwidth to work on it)
